openfecli/commands/gather.py

```diff
@@ -78,26 +78,63 @@ def is_results_json(fpath:os.PathLike|str)->bool:
     """Sanity check that file is a result json before we try to deserialize"""
     return 'estimate' in open(fpath, 'r').read(20)
 
-
-def load_results(fpath:os.PathLike|str)->dict:
-    """Load the data from a results JSON into a dict
+def load_json(fpath:os.PathLike|str)->dict:
+    """Load a JSON file containing a gufe object.
 
     Parameters
     ----------
     fpath : os.PathLike | str
-        The path to deserialized results.
+        The path to a gufe-serialized JSON.
+
 
     Returns
     -------
     dict
         A dict containing data from the results JSON.
-    """
 
+    """
+    # TODO: move this function to openfe/utils
     import json
     from gufe.tokenization import JSON_HANDLER
 
     return json.load(open(fpath, 'r'), cls=JSON_HANDLER.decoder)
 
+def load_valid_result_json(fpath:os.PathLike|str)->dict|None:
+    """Load the data from a results JSON into a dict.
+
+    Parameters
+    ----------
+    fpath : os.PathLike | str
+        The path to deserialized results.
+
+    Returns
+    -------
+    dict | None
+        A dict containing data from the results JSON,
+        or None if the JSON file is invalid or missing.
+
+    """
+
+
+    try:
+        result = load_json(fpath)
+    except FileNotFoundError:
+        click.echo(f"Warning: {fpath} does not exist. Skipping.", err=True)
+        return None
+    if "unit_results" not in result.keys():
+        click.echo(f"{fpath}: No 'unit_results' found, assuming to be a failed simulation.", err=True)
+        return None
+    if result['estimate'] is None:
+        click.echo(f"{fpath}: No 'estimate' found, assuming to be a failed simulation.", err=True)
+        return None
+    if result['uncertainty'] is None:
+        click.echo(f"{fpath}: No 'uncertainty' found, assuming to be a failed simulation.", err=True)
+        return None
+    if all('exception' in u for u in result['unit_results'].values()):
+        click.echo(f"{fpath}: Exception found in all 'unit_results', assuming to be a failed simulation.", err=True)
+        return None
```

Comment:
few notes/ideas
1. I think using the [`match`](https://docs.python.org/3/reference/compound_stmts.html#overview) statement for all the `if` checks would be cleaner
2. Do we need to worry about the case of `exception`? I'm not sure if we need a `.lower()` call or if we have an agreement with protocol devs that they ensure `exception` will be in the values of `unit_results` if there is one

Reply:
2. in this case, `openfe gather` is only to be used for protocols built-in to our CLI, so I think this is fine as-is for now. When we want to make this more general later, I'm thinking we'll want to update this function anyway.

Reply:
1. i'm open to using a match statement, but I'm not sure how to write that for the first and last checks. I can look into it and if it's cleaner add it in a follow-up